matrix_sdk_crypto/src/machine.rs

```diff
@@ -932,10 +928,30 @@ impl OlmMachine {
             .get(&KeyAlgorithm::Ed25519)
             .ok_or(EventError::MissingSigningKey)?;
 
-        Ok((
-            serde_json::from_value::<EventJson<ToDeviceEvent>>(decrypted_json)?,
-            signing_key.to_owned(),
-        ))
+        let ev_type = decrypted_json
+            .get("type")
+            .cloned()
+            .ok_or_else(|| EventError::MissingField("type".to_string()))?;
+
+        let content = decrypted_json
+            .get("content")
+            .cloned()
+            .ok_or_else(|| EventError::MissingField("content".to_string()))?;
+
+        let sender = decrypted_json
+            .get("sender")
+            .cloned()
+            .ok_or_else(|| EventError::MissingField("sender".to_string()))?;
+
+        let device_event = json!({
+            "type": ev_type,
+            "content": content,
+            "sender": sender
+        });
+
+        let event = serde_json::from_value::<EventJson<AnyToDeviceEvent>>(device_event)?;
```

Comment:
You answered this while I was doing the review, the deserialization should not fail inside of Ruma, additional fields should always be ignored. I'm a bit confused on why this isn't so already since you need to request from serde to fail using [this](https://serde.rs/container-attrs.html#deny_unknown_fields) flag.

Reply:
Ahhh :facepalm: dang this is a custom `Deserialize` impl I can't believe I spent so long trying to figure this out. So I guess I'll have to change that in ruma.

Reply:
Serde's example is what the event structs Deserialize is based on and it fails when it encounters an unknown filed [we used the commented out derive](https://serde.rs/deserialize-struct.html) and that is where it fails.